src/deluge/dsp/timestretch/time_stretcher.cpp

```diff
@@ -979,6 +979,8 @@ bool TimeStretcher::hopEnd(SamplePlaybackGuide* guide, VoiceSample* voiceSample,
 bool TimeStretcher::setupNewPlayHead(Sample* sample, VoiceSample* voiceSample, SamplePlaybackGuide* guide,
                                      int32_t newHeadBytePos, int32_t additionalOscPos, int32_t priorityRating,
                                      LoopType loopingType) {
+	//clear the current reasons since setting up the clusters will add new ones
+	voiceSample->unassignAllReasons();
```

Comment:
Should we do this right at the end and then re-add a single reason? Otherwise there's a window if time where a cluster we're about to need could get stolen, I think.

Reply:
The `setupClusterForPlayFromByte` logic does go right in to assigning reasons so it's probably OK, but it might be worth leaving a note.

Reply:
Will do - I think it's ok right now since clusters can't be allocated by the interrupts but a warning is a good idea

Reply:
Done